### DISCUSSION TITLE
Discord.py @tasks.loop equivalent?

### DISCUSSION THREAD

**Discussion by vinicius507** [andersfylling/disgord#434] [Q&A] [answered]
Greetings,

I'm refactoring a `discord.py` Discord bot I have using `disgord`. In it I have some task loops that I'd like to recreate. Is there any built-in method in `disgord` to create event loops/schedule tasks?

Thanks in advance,
Myriad

**Comment by andersfylling** [comment, marked as answer]
in python this is just coroutines scheduling if I'm correctly understanding you. Disgord allows you to register a channel for each event type, you can then have a goroutine dedicated for each to process events: https://pkg.go.dev/github.com/andersfylling/disgord#hdr-Listen_for_events_using_Channels


outdated docs here (the client.On method no longer exists so you'd use client.Gateway().*Chan(...) instead):
https://github.com/andersfylling/disgord/blob/master/docs/examples/listen-for-messages.md

**Reply by vinicius507** [reply]
Yes, exacly. I was just asking in case there was a built-in feature to enable coroutines. The coroutines I need to build aren't related to Discord Events. E.g.:  a coroutine that sets up an user role depending on a event outside of Discord.

**Reply by andersfylling** [reply]
sounds like a general Golang question?
to make a function a coroutine you just use "go" when you call it:
```go
go func() {
    // do smth in a co routine
}
```

so essentially you would write your own goroutine and call the required disgord calls to setup the user. All REST requests are blocking, and there is no way to make them "async". You just have to handle it in a goroutine yourself.

**Reply by vinicius507** [reply]
Awesome, thanks.
